app/plugin_commands.go

```diff
@@ -18,7 +20,27 @@ type PluginCommand struct {
 
 func (a *App) RegisterPluginCommand(pluginId string, command *model.Command) error {
 	if command.Trigger == "" {
-		return fmt.Errorf("invalid command")
+		return errors.New("invalid command")
+	}
+	if command.AutocompleteData != nil && command.AutocompleteData.IsValid() != nil {
+		return errors.New("invalid autocomplete data in command")
+	}
+
+	if command.AutocompleteData == nil {
+		command.AutocompleteData = model.NewAutocompleteData(command.Trigger, command.AutoCompleteHint, command.AutoCompleteDesc)
+	} else {
+		siteURL := a.GetSiteURL()
+		if siteURL == "" {
+			siteURL = fmt.Sprintf("http://localhost%v", *a.Srv().Config().ServiceSettings.ListenAddress)
+		}
+		baseURL, err := url.Parse(siteURL + "/plugins/" + pluginId)
+		if err != nil {
+			return errors.Wrapf(err, "Can't parse url %s", siteURL+"/plugins/"+pluginId)
+		}
+		err = command.AutocompleteData.UpdateRelativeURLsForPluginCommands(baseURL)
+		if err != nil {
+			return errors.Wrap(err, "Can't update relative urls for plugin commands")
+		}
```

Comment:
Why are you dealing with the relativeURL vs. absoluteURL problem on the server side? It seams easier to me to just have the webapp build the request URL itself, if it starts with `/plugins/`.

Reply:
I don't like it either. This method is not called from the webapp, it's called from the server itself to fetch dynamic arguments from the plugin.

Reply:
Got it. Does it make sense to move the handling to https://github.com/mattermost/mattermost-server/pull/14259/files#diff-02aa38f226a808932a2b1e34a8cc834fR116 and do something like https://github.com/mattermost/mattermost-server/blob/master/app/integration_action.go#L270-L273?